### ISSUE TITLE
MKL 2021 version with conda setup not working

### ISSUE THREAD

**Issue by neiladit** [flatironinstitute/sparse_dot#15]
Hi,

In my understanding, the wrapper tries to find the file `mkl_rt.dll` but the MKL installation on conda (2021.2.0) does not have any file with that name. The closest I could find is `mkl_rt.1.dll`. I get the following error when I try to `import sparse_dot_mkl` : 
`ImportError: mkl_rt not found.` Have you faced this issue? Maybe this is Windows specific?

Thanks for helping out!

**Comment by asistradition** [comment]
I'll look into it, although I'd be very surprised if MKL changed the library name intentionally. (It would have to be windows specific as DLLs are windows libraries)

Can you provide the conda environment yaml?

**Comment by neiladit** [comment]
@asistradition Thanks for looking into this! Fwiw it worked on Ubuntu for me. Here is the environment yaml on Windows:
```
name: test
channels:
  - conda-forge
  - intel
  - defaults
dependencies:
  - blas=1.0=mkl
  - ca-certificates=2020.12.5=0       
  - certifi=2021.5.30=py39haa95532_0  
  - intel-openmp=2021.2.0=haa95532_616
  - libprotobuf=3.16.0=h7755175_0     
  - mkl=2021.2.0=intel_296
  - mkl-service=2.3.0=py39h2bbff1b_1  
  - mkl_fft=1.3.0=py39h277e83a_2      
  - mkl_random=1.2.1=py39hf11a4ad_2   
  - numpy=1.20.2=py39ha4e8547_0       
  - numpy-base=1.20.2=py39hc2deb75_0  
  - pip=21.1.2=py39haa95532_0
  - python=3.9.5=h6244533_3
  - python_abi=3.9=1_cp39
  - setuptools=52.0.0=py39haa95532_0
  - six=1.15.0=py39haa95532_0
  - sqlite=3.35.4=h2bbff1b_0
  - tbb=2021.2.0=vc14_intel_371
  - typing-extensions=3.10.0.0=hd8ed1ab_0
  - typing_extensions=3.10.0.0=pyha770c72_0
  - tzdata=2020f=h52ac0ba_0
  - vc=14.2=h21ff451_1
  - vs2015_runtime=14.27.29016=h5e58377_2
  - wheel=0.36.2=pyhd3eb1b0_0
  - wincertstore=0.2=py39h2bbff1b_0
  - zlib=1.2.11=h62dcd97_1010
  - pip:
    - onnx==1.9.0
    - pandas==1.2.4
    - protobuf==3.17.3
    - python-dateutil==2.8.1
    - pytz==2021.1
    - regex==2021.4.4
    - scipy==1.6.3
    - sparse-dot-mkl==0.7.1
   ```
I tried both Intel and conda channels for `mkl` library.

**Comment by asistradition** [comment]
Awesome; it's not just you, I can replicate this. Looks like the mkl dll is indeed renamed from `mkl_rt.dll` to `mkl_rt.1.dll` in the newest conda release of mkl 2021.

The fix is trivial (this isn't the only package affected: https://github.com/brightway-lca/brightway2-calc/issues/36) so I'll try to get it done ASAP.

Thanks!

**Comment by asistradition** [comment]
Looks like `mkl_rt.1.so` is the new name for the oneMKL libraries; the osx and linux packages all symlink `mkl_rt.so` to it so there's no issues, but windows doesn't and of course the library pattern-matching in `ctypes.utils.find_library` doesn't work on windows either.

Pushed a fix in 86228d0d7b5d7dc728426dc5eec1adba15e5a7c8 and released it as v0.7.2.

Thanks!
